src/components/Api.js

```diff
@@ -0,0 +1,153 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+    // constructor body
+  }
+
+  async getInitialCards() {
+    try {
+      const res = await fetch(`${this._baseUrl}/cards`, {
+        method: "GET",
+        headers: this._headers,
+      });
+      if (!res.ok) {
+        throw new Error(`HTTP error! Status: ${res.status}`);
+      }
+      return await res.json();
+    } catch (err) {
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.

Reply:
how do i remove catch. because when i tried its marking the code red.

Reply:
> how do i remove catch. because when i tried its marking the code red.

You don't need `try/catch`. It should be just like this

```js

  getInitialCards() {
    return fetch(`${this._baseUrl}/cards`, {
      method: "GET",
      headers: this._headers,
    }).then(this.checkResponse);
  }

```